### ISSUE TITLE
locale uses deprecated geoip endpoint (ends 2018-07-01)

### ISSUE THREAD

**Issue by apachelogger** [calamares/calamares#929]
#### Submission type

  - [x] Bug report
  - [ ] Feature Request

locale module has

```
QString requestUrl = QString( "%1/json" )
                         .arg( QUrl::fromUserInput( m_geoipUrl ).toString() );
```

which results in 

```
➜ curl freegeoip.net/json
<a href="/json/">Moved Permanently</a>.
```

and then

```
➜ curl freegeoip.net/json/
{"__deprecation_message__":"This API endpoint is deprecated and will stop working on July 1st, 2018. For more information please visit: https://github.com/apilayer/freegeoip#readme",........}
```

**This API endpoint is deprecated and will stop working on July 1st, 2018. For more information please visit: https://github.com/apilayer/freegeoip#readme**

**Comment by abucodonosor** [comment]
Well I got a patch some time ago to allow custom geoip services since now we 
hardcode to freegeoip.. None wanted that so I just dropped it.

**Comment by adriaandegroot** [comment]
This is available in 3.1.13 (released now) and from 3.2-rc5 onwards (some time next week). Handles the legacy (weird-hard-coded, as @abucodonosor  says) case, as well as JSON-style data and XML-style data (Ubiquity).
